src/main/java/com/intuit/graphql/orchestrator/batch/GraphQLServiceBatchLoader.java

```diff
@@ -73,7 +73,24 @@ private GraphQLServiceBatchLoader(Builder builder) {
   @SuppressWarnings("rawtypes")
   @Override
   public CompletionStage<List<DataFetcherResult<Object>>> load(final List<DataFetchingEnvironment> keys) {
-    GraphQLContext context = getContext(keys);
+    GraphQLContext graphQLContext = getContext(keys);
+    FieldAuthorization<AuthDataT> fieldAuthorization = getFieldAuthorizationFromOrDefault(graphQLContext);
+    if (Objects.nonNull(fieldAuthorization)) {
+      CompletableFuture<AuthDataT> futureAuthData = fieldAuthorization.getFutureAuthData();
+      return futureAuthData.thenCompose(authData -> load(keys, authData, graphQLContext, fieldAuthorization));
```

Comment:
need to further optimize this as this always tries to complete the future even if the the data fetching environment selection set does not require authorization.